crates/matrix-sdk/src/send_queue.rs

```diff
@@ -1888,6 +1914,9 @@ pub struct SendHandle {
 
     /// Additional handles for a media upload.
     media_handles: Option<MediaHandles>,
+
+    /// The time that this send handle was first created
+    pub created_at: Option<MilliSecondsSinceUnixEpoch>,
```

Comment:
I see that the only place where this is read is in the timeline code, correct? As a last high-level comment: I think this could be moved to the `LocalEchoContent::Event` variant, even — we might not care about a reaction's `created_at` (or do we? if so, we could move it to the `LocalEcho` struct more generally).

Reply:
So the issue is that it needs to be exposed in the `LocalEventTimelineItem` in `crates/matrix-sdk-ui/src/timeline/event_handler.rs` which doesn't appear to have access to LocalEchos.  Thats why I put it in the SendHandle.